### ISSUE TITLE
Disabling Line Length Checks

### ISSUE THREAD

**Issue by cadojo** [rvben/rumdl#121]
Thanks for the great project!

Is it possible to disable `line_length` entirely, especially if we have `sentence-per-line` enabled? After updating to the latest version, I've noticed checks failing due to a default line length of 80 characters. I don't have any line length set in my configuration file, and I have my reflow mode set to sentence-per-line.

Setting `line_length = -1` solves the issue, but is that officially supported? Is there another syntax I can use to allow all line lengths? The schema shows `line_length` should be unsigned.

Relates: https://github.com/rvben/rumdl/issues/111

**Comment by rvben** [comment]
Hi @cadojo,

Yeah, I guess you could say that setting `line-length = -1` is a hack, but it works.. :)
To be consistent with other options in MD013, I've added the `paragraphs` key.

Configuration example:

```yaml
MD013:
  paragraphs: false  # Don't warn about long paragraphs
  reflow: true
  reflow-mode: "sentence-per-line"
```

In that case, it will not complain anymore about line length for paragraphs (including sentences). This will be available in 0.0.166.

**Comment by jdc-pub** [comment]
Thank you! I'm curious, could you clarify what kind of line could exceed the `line-length if `paragraphs: true`? I can't seem to think of a line in Markdown that wouldn't be considered a paragraph. For example, bullet points can be thought of as paragraphs (I think). 

In GitHub issues, multi-sentence bullet points resolve newlines. But in GitHub's rendering of README files, for example, multi-sentence bullet points treat newlines as spaces in a paragraph.

**Comment by rvben** [comment]
Hi @cadojo,

Great question. Paragraphs are all text that's no structural/special content.

The key distinction is:

  - Structural/special elements = headings, code blocks, tables, blockquotes, HTML
  - Paragraph-like content = regular text and list items (skipped when paragraphs: false)

Does that clarify things? (And does it make sense? I'm open for feedback/ideas)

**Comment by jdc-pub** [comment]
That does make sense! I really like keeping one-sentence-per-line for cleaner Git diffs, and so I'd love to disable line-length checks entirely. I think an officially supported option there, including for structural / special elements, would be valuable.

Is there room for `line-length: 0` or `line-length: null` or `line-length: false` to disable checks? Or is there room to officially support `line-length: -1`? (I'm guessing that's causing an overflow in the code, so maybe the `-1` shouldn't be officially supported?)

**Comment by rvben** [comment]
Hi @cadojo,

Thanks for being persistent. I really appreciate it (because it shows you care). After consideration it indeed makes most sense to just support line-length of 0 to mean 'unlimited'. That makes it way easier to disable it, instead of having to disable all the different types of content. Now `line-length=0` will be enough to disable it completely.

I'll make this part of next release. Thanks again for making rumdl more user-friendly and support more usecases! 🚀

**Comment by jdc-pub** [comment]
Thank you for this awesome project!
